packages/react/src/hooks/useFormatSelection.js

```diff
@@ -0,0 +1,40 @@
+const useFormatSelection = (messageRef) => {
+  const formatSelection = (pattern) => {
```

Comment:
can we separate out this function to a file and add some basic unit tests to it if possible?

Reply:
It's already in a different file Sid, regarding unit test, I'll add it later.